plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/PositionDeleteFilter.java

```diff
@@ -0,0 +1,121 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.delete;
+
+import io.airlift.slice.Slice;
+import io.trino.plugin.iceberg.IcebergColumnHandle;
+import io.trino.spi.Page;
+import io.trino.spi.block.Block;
+import io.trino.spi.connector.ConnectorPageSource;
+import org.apache.iceberg.MetadataColumns;
+import org.apache.iceberg.Schema;
+import org.roaringbitmap.longlong.ImmutableLongBitmapDataProvider;
+import org.roaringbitmap.longlong.LongBitmapDataProvider;
+import org.roaringbitmap.longlong.Roaring64Bitmap;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public final class PositionDeleteFilter
+        implements DeleteFilter
+{
+    private final ImmutableLongBitmapDataProvider deletedRows;
+
+    private PositionDeleteFilter(ImmutableLongBitmapDataProvider deletedRows)
+    {
+        this.deletedRows = requireNonNull(deletedRows, "deletedRows is null");
+    }
+
+    @Override
+    public Schema schema()
+    {
+        return new Schema(MetadataColumns.ROW_POSITION);
+    }
+
+    @Override
+    public RowPredicate createPredicate(List<IcebergColumnHandle> columns)
+    {
+        int filePosChannel = rowPositionChannel(columns);
+        return (page, position) -> {
+            long filePos = BIGINT.getLong(page.getBlock(filePosChannel), position);
+            return !deletedRows.contains(filePos);
+        };
+    }
+
+    private static int rowPositionChannel(List<IcebergColumnHandle> columns)
+    {
+        for (int i = 0; i < columns.size(); i++) {
+            if (columns.get(i).isRowPositionColumn()) {
+                return i;
+            }
+        }
+        throw new IllegalArgumentException("No row position column");
+    }
+
+    public static DeleteFilter readPositionDeletes(ConnectorPageSource pageSource, Slice targetPath)
+    {
+        LongBitmapDataProvider deletedRows = new Roaring64Bitmap();
+        LazyVarcharComparator comparator = new LazyVarcharComparator(targetPath);
+
+        while (!pageSource.isFinished()) {
+            Page page = pageSource.getNextPage();
+            if (page == null) {
+                continue;
+            }
+
+            Block pathBlock = page.getBlock(0);
+            Block posBlock = page.getBlock(1);
+
+            for (int position = 0; position < page.getPositionCount(); position++) {
+                int result = comparator.compare(pathBlock, position);
+                if (result > 0) {
+                    return new PositionDeleteFilter(deletedRows);
+                }
+                if (result == 0) {
+                    deletedRows.addLong(BIGINT.getLong(posBlock, position));
```

Comment:
Document why doing linear search for first match instead of eg bisecting
(eg deletion files are typically created for single data file, so doing linear search here is OK)

Reply:
I didn't even consider a binary search, since that seems complicated to write.

Reply:
Plus, the predicate passed to the PageSource is already filtering out row groups / stripes that don't have the targetPath in them